processes/ingest/hathi_trust.py

```diff
@@ -99,23 +107,14 @@ def importFromHathiFile(self, hathiURL):
             self.readHathiFile(hathiTSV)
 
     def readHathiFile(self, hathiTSV):
-        processed_record_count = 0
-
-        while True:
-            if self.ingest_limit and processed_record_count >= self.ingest_limit:
+        for number_of_books_ingested, book in enumerate(hathiTSV):
+            if self.ingest_limit and number_of_books_ingested > self.ingest_limit:
                 break
 
-            try:
-                row = next(hathiTSV)
-                logger.info('Reading row')
-            except csv.Error as e:
-                logger.warning('Unable to read TSV row')
-                logger.debug(e)
-                continue
-            except StopIteration:
-                logger.info('Reached end of TSV file')
-                break
+            book_right = book[2]
+            book_date_updated = book[14]
```

Comment:
Recommend guarding against index out of bounds exceptions by checking the length of the array first.

Reply:
That's a good check to make to avoid that exception.